### ISSUE TITLE
Deprecate SetNameAndDescription()

### ISSUE THREAD

**Issue by mast-eu** [gitextensions/gitextensions.plugintemplate#5]
@RussKie raised an issue that the scope of the method `SetNameAndDescription(name)` in `GitPluginBase` is not very clear: https://github.com/gitextensions/gitextensions.plugintemplate/pull/1#discussion_r291406400.

I did some tests with the Statistics plugin to figure it out. I replaced `SetNameAndDescription("Statistics");` with 
```
Name = "Statistics (name)";
Description = "Statistics (description)";
```

_Description_ is a localized string that is used in both the plugins main menu and settings:

![image](https://user-images.githubusercontent.com/46861028/59100969-e3cb6b80-8927-11e9-83ce-2cf6574438fa.png)

![image](https://user-images.githubusercontent.com/46861028/59101657-f2b31d80-8929-11e9-9c2e-b97c9959cbd2.png)

_Name_ is used as prefix in `GitExtensions.settings`. This is not localized, so settings do not get lost when user switches language.

```
  <item>
    <key>
      <string>Statistics (name)Directories to ignore (EndsWith)</string>
    </key>
    <value>
      <string>\Debug;\Release;\obj;\bin</string>
    </value>
  </item>
```

Using the _Name_ as prefix already caused problems in the past and there's an open issue about this: https://github.com/gitextensions/gitextensions/issues/5797.

I think we should substitute _Name_ with a _GUID_ (as suggested in https://github.com/gitextensions/gitextensions/issues/5797) and rename  _Description_ to _Name_.

Afterwards, `SetNameAndDescription()` can be removed.

**Comment by RussKie** [comment]
Sounds good 👍 

I would also limit the `Name` property (the one that is `Description` now) to something reasonable (like 20 or 50 chars).
We don't want a crazily long plugin name

**Comment by mast-eu** [comment]
For sure we should limit the name length. 60 chars should be fine.
The worst case right now is "Azure DevOps and Team Foundation Server (since TFS2015)" (55 chars).

**Comment by RussKie** [comment]
> The worst case right now is "Azure DevOps and Team Foundation Server (since TFS2015)" (55 chars).

I'd argue that should be a description (or a tooltip) but not a name. A name should be something like "_Azure DevOps / TFS_"

**Comment by mast-eu** [comment]
I agree. Names will be cheap to change once we're using GUIDs for the settings.
